### ISSUE TITLE
Add alertdialog to non-dismissible modals

### ISSUE THREAD

**Issue by arodenbeck** [salesforce/design-system-react#1237]
Pprompt variant does not close with escape

Generally modals have a close button, which can either be activated using the space or enter when focused, or by hitting escape. This variant does not have a close button, but should still close on escape.

**Comment by donnieberg** [comment]
@arodenbeck @interactivellama There's a prop called `dismissible` and if it's false, then the user can't press `esc` to close the modal because they're expected to do something inside the modal to close it. It's set to true by default. Is this ok if the developer wants to force a user to interact with the modal (ie. a prompt that has an important error message or terms and conditions where you want the user to explicitly click "ok")?

**Comment by interactivellama** [comment]
I don't see `role="alertdialog"` anywhere in the HTML. Would that help to say there is no esc or close button?

**Comment by arodenbeck** [comment]
@interactivellama changing the role from dialog to alertdialog does make sense in this case. It more than likely would not make any difference today, as it seems to be up to the browser and AT how they treat alertdialogs. I think it is forward thinking to go ahead and add this.

**Comment by interactivellama** [comment]
@donnieberg Consensus is make sure `alertdialog` is in there and then we can close this issue. Does that sound good to you?

I'd recommend building on top of https://github.com/salesforce/design-system-react/pull/1391 to avoid merge conflicts.

**Comment by donnieberg** [comment]
that sounds like a plan! Thank you for your input @arodenbeck and @interactivellama!

**Comment by donnieberg** [comment]
@arodenbeck or @interactivellama would one of you be able to check out my PR onto @davidlygagnon 's branch? https://github.com/davidlygagnon/design-system-react/pull/2

**Comment by davidlygagnon** [comment]
I just merged your branch into the modal upgrade PR.

**Comment by interactivellama** [comment]
Fixed in #1391
